### DISCUSSION TITLE
microproが反応しない

### DISCUSSION THREAD

**Discussion by poporirinu** [Yowkees/keyball#543] [Q&A]
マイコン単体だと光ります
が基盤に差し込むと反応しません。
左右のマイコンを入れ替えても同様です。
ハンダの温め直しなどは行いましたが変化なしです。
恐れ入りますがご回答お願い致します。
![image](https://github.com/Yowkees/keyball/assets/160295919/57b15f21-27c4-464f-b9ac-dd2ef9d5f14a)![image](https://github.com/Yowkees/keyball/assets/160295919/69448188-01fa-4c48-950f-f88353cc5363)![image](https://github.com/Yowkees/keyball/assets/160295919/3ee2f51b-bc11-4bd9-9277-a54c7084bc58)

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

添付させていただいた写真の赤丸で示したProMicroの端子のはんだが隣の端子に触れているように見えますので、こちらを除去してみてください。
他にも隣の端子にはんだが触れていないかの確認をお願いいたします。
![319666150-57b15f21-27c4-464f-b9ac-dd2ef9d5f14a](https://github.com/Yowkees/keyball/assets/143698349/01b3c353-2204-4c83-80d4-7c7bd09f039f)

**Reply by poporirinu** [reply]
ありがとうございます。無事治りました。

別件ですが16からLEDが点灯いたしません。
こちらもご確認お願い致します![image](https://github.com/Yowkees/keyball/assets/160295919/88cc7f0f-d694-490b-9888-41c5207b5fd7)![image](https://github.com/Yowkees/keyball/assets/160295919/6c436955-d3d5-49ac-b0e4-bc7961eba0cc)

**Reply by Ryo624** [reply]
ご対応、解決していただきましてありがとうございます。
LEDが点灯しない場合は、点灯しないLEDかそのひとつ前（今回の場合は15番）のLEDのはんだ付けを見直してみてください。
LEDは熱にとても弱いので、はんだごての温度、作業時間に気を付けてください。

**Reply by poporirinu** [reply]
ありがとうございます。
無事点灯致しました。

お忙しいところ迅速なご対応ありがとうございました。

すみません、追加でご確認お願い致します。
test matrixの
形状が現物と異り、トラックボールが認識されない？のですが
こちらはどのように対応すればよろしいでしょうか![image](https://github.com/Yowkees/keyball/assets/160295919/9b7f1c48-238c-4b88-8f4d-7996c7a1b4b2)![image](https://github.com/Yowkees/keyball/assets/160295919/48de9478-6a45-4abd-b4c2-d93ed06f1580)

また、特定のキーだけ反応しません。対応策をご教示願います
![image](https://github.com/Yowkees/keyball/assets/160295919/6e65f270-92d0-4343-a2f5-9640a34ff6cc)
![image](https://github.com/Yowkees/keyball/assets/160295919/f2542a97-2621-4fdb-b3ee-c7c9e8dc169d)

**Reply by Ryo624** [reply]
トラックボール基板を差し込むことでトラックボール有り側の基板（右手側）と認識されますので、トラックボール基板を差し込んでからもう一度Remapを開いてみてください。

反応のないキーですが、一つのキーが効かないということですので、キーソケットもしくはダイオードが原因となっている可能性が高いです。まずはキーソケットのはんだ付けをやり直していただいて、改善されなければダイオードのはんだ付けもやり直してみてください。
今回のキーに対応するダイオードはすぐ上のダイオードになります。

**Reply by poporirinu** [reply]
ありがとうございます。

特定のキーが反応しない件について
ご教示頂きました内容を行いましたが改善されませんでした。基盤の画像を添付致しますのでご確認お願い致します![image](https://github.com/Yowkees/keyball/assets/160295919/4972299d-7fe8-4e87-aeae-cdc275acb567)

また、まるで囲んだ部品は交換致しました
![image](https://github.com/Yowkees/keyball/assets/160295919/30e38012-3e89-43ce-866c-e927b4977a2c)

![image](https://github.com/Yowkees/keyball/assets/160295919/99a62e8d-b776-4a36-95db-f26166e87705)

**Reply by Ryo624** [reply]
ご対応ありがとうございます。部品の交換も行ったということで了解しました。

可能性は低いかもしれませんが添付しました写真の青い丸の端子のはんだ付けを他の端子のようにはんだの形が山型になるようにはんだ付けをやり直してみてもよろしいでしょうか。

![319868832-99a62e8d-b776-4a36-95db-f26166e87705](https://github.com/Yowkees/keyball/assets/143698349/66c34d63-e89b-43e9-9cb0-84cf5264ee81)

**Reply by poporirinu** [reply]
ご回答ありがとうございます

ハンダを山形になるように対応致しましたが改善されませんでした。![image](https://github.com/Yowkees/keyball/assets/160295919/1586ca6b-5a50-4c6a-9d81-8fabf93c5955)

**Reply by Ryo624** [reply]
ご対応ありがとうございます。

キースイッチを別のものに変えても症状に変わりはないでしょうか？

根気よくやり直しが必要になる場合もありますので、ソケットと端子の間にしっかりはんだが馴染むよう何度かキーソケットのはんだ付けをやり直してみてください。
はんだの量が多くならないように注意してください。

**Reply by poporirinu** [reply]
ありがとうございます。
確認してみます。

COLの導通チェックをしたのですが
反応しないキーと同箇所の黄色丸の2箇所、反応がみられず
ピンソケットのランド剥がれが見られました。
こちらは修復可能でしょうか。
何度も申し訳ございません。![image](https://github.com/Yowkees/keyball/assets/160295919/f3d28672-da27-4116-83a7-66fa9359a69b)![image](https://github.com/Yowkees/keyball/assets/160295919/70f3539f-ebec-468b-a20a-1fc522d5a80d)

**Reply by poporirinu** [reply]
お世話になっております。
昨日はご教示頂きありがとうございました。
白銀ラボへ右基盤を購入したい旨等の問い合わせをさせていただいたのですが届いておりますでしょうか？もし届いていない様でしたら再度送らせて頂きますので教えて下さい、よろしくお願い致します
